lib/service/state.go

```diff
@@ -27,23 +27,25 @@ import (
 
 const (
 	// stateOK means Teleport is operating normally.
-	stateOK = 0
-
+	stateOK = iota
 	// stateRecovering means Teleport has begun recovering from a degraded state.
-	stateRecovering = 1
-
+	stateRecovering
 	// stateDegraded means some kind of connection error has occurred to put
 	// Teleport into a degraded state.
-	stateDegraded = 2
+	stateDegraded
+	// stateStarting means the process is starting but hasn't joined the
+	// cluster yet.
+	stateStarting
 )
 
 var stateGauge = prometheus.NewGauge(prometheus.GaugeOpts{
 	Name: teleport.MetricState,
-	Help: "State of the teleport process: 0 - ok, 1 - recovering, 2 - degraded",
+	Help: "State of the teleport process: 0 - ok, 1 - recovering, 2 - degraded, 3 - starting",
```

Comment:
NIT: may be nicer to define this help text with a `fmt.Sprintf` so that the values of the constants can be a "single source of truth".  I don't know why we'd ever change them, but its always good to have changes propagate automatically.